tests/core/utilities/test_decorators.py

```diff
@@ -0,0 +1,22 @@
+import time
+
+from web3.utils.threads import (
+    spawn,
+)
+from web3.utils.decorators import (
+    reject_recursive_repeats,
+)
+
+
+def test_reject_recursive_repeats_multithreaded():
+    @reject_recursive_repeats
+    def recurse(sleep_now):
+        time.sleep(sleep_now)
+        try:
+            recurse(0.05)
+            return True
+        except ValueError:
+            return False
+    thd1 = spawn(recurse, 0)
+    thd2 = spawn(recurse, 0.02)
+    assert thd2.get() and thd1.get()
```

Comment:
*curious* 

When you tested this, did it trigger the error 100% of the time?

Reply:
I tested it 3 or 4 times, and it triggered every time. I'll go run a quick test to make sure it triggers at least 200 times in a row.

Reply:
On my laptop, I can trigger this failure 500 times in a row without the thread-aware decorator.